xc/common/utils/prjxray_create_place_constraints.py

```diff
@@ -361,10 +361,9 @@ def __init__(
                 clock = self.clock_blocks[cname]
 
                 for sink in sinks:
-                    assert sink in ports, (
-                        cname,
-                        sink,
-                    )
+                    if sink not in ports:
```

Comment:
I don't believe this change is right?

Reply:
There are multiple reasons for this:

- apparently there was a change that caused all the `$false` or `$true` nets of bus ports to be named with the full name bus instead of the single bit:
  - e.g. PS7 example: `FCLKCLKTRIGN=$false` where the `FCLKCLKTRIGN` port is a multi-bit bus. This causes issues with https://github.com/SymbiFlow/symbiflow-arch-defs/blob/098fe2e8e724bca3902b8121fb254caaed7cb5d3/xc/common/utils/prjxray_create_place_constraints.py#L57-L60.

The above can be actually treated in a better way, and I'll address it.

- with the GTP_COMMON, there is a situation in which either the `GTREFCLK0` or `GTREFCLK1` will be present in the blif.
Given that the GTP_COMMON is a clock-related tile which needs to be tightly coupled with a corresponding IBUFDS_GTE2, we need to add it to the CLOCKS dict, but the fact is that, to avoid impossible routes to route GND, to [the unused GTREFCLK are dropped](https://github.com/SymbiFlow/symbiflow-arch-defs/blob/3b3a14a8864ea9c9c20688cf76b1b3aeaf03080b/xc/xc7/techmap/cells_map.v#L8428-L8633).

Reply:
> apparently there was a change that caused all the $false or $true nets of bus ports to be named with the full name bus instead of the single bit:

This sounds like a bug in the yosys blif writer?

Reply:
Rather than the blif writer, this is caused by something else, I could not identify what yet though.

The top_synth.json has a difference regarding ports connected to GND:

Old yosys version:
```
...
"FCLKCLK": [ 476, 477, 478, 479 ],
"FCLKCLKTRIGN": [ "0", "0", "0", "0" ],
"FCLKRESETN": [ 480, 481, 482, 483 ], 
...
```

New yosys version:
```
...
"FCLKCLK": [ 476, 477, 478, 479 ],
"FCLKCLKTRIGN": [ "0" ],
"FCLKRESETN": [ 480, 481, 482, 483 ], 
...
```

Reply:
I would take a look at the cell definition (e.g. `cells_sim.v`).

Reply:
In yosys, the PS7 definition did not change between the two versions (in the [cells_xtra.v](https://github.com/YosysHQ/yosys/blob/master/techlibs/xilinx/cells_xtra.v)) and neither nothing changed in arch-defs as far. (The PS7 is just an example here, I should verify for other primitives as well).

I shall also run bisect probably to see what change is causing this behaviour.

Reply:
After some debugging, I discovered the reason behind this. Basically, it corresponds to a techmap issue with the PS7 block around the handling of unconnected ports, so this is fixable in arch-defs.